### ISSUE TITLE
Uncaught TypeError: Cannot read property 'isPreview' of undefined on saving

### ISSUE THREAD

**Issue by MatzeKitt** [WordPress/gutenberg#29711]
## Description
While trying to save a post with changed reusable blocks, I sometimes get the following error message after clicking on “Update” and then “Save”:

```
Uncaught TypeError: Cannot read property 'isPreview' of undefined
    at Te (editor.min.js?ver=5210dd1dd91e796c406f0aacfa4b9620:12)
    at data.min.js?ver=943087ae96d075f126df689839bb96b9:2
    at Object.r [as isPreviewingPost] (data.min.js?ver=943087ae96d075f126df689839bb96b9:2)
    at f (gbc.min.js?ver=4.1.3:1)
    at data.min.js?ver=943087ae96d075f126df689839bb96b9:2
    at Array.forEach (<anonymous>)
    at i (data.min.js?ver=943087ae96d075f126df689839bb96b9:2)
    at data.min.js?ver=943087ae96d075f126df689839bb96b9:2
    at p (data.min.js?ver=943087ae96d075f126df689839bb96b9:2)
    at redux-routine.min.js?ver=284288e957394d2097c6fbe95625e2fb:2
```

It does in fact always work in Safari.

## Step-by-step reproduction instructions
Since it doesn’t occur always, it’s hard to find a way to always reproduce it. If I reload the edit page, remove all reusable blocks from the post, save it and add the reusable blocks again, it’s working well.

## Expected behaviour
The post and all reusable blocks gets saved.

## Actual behaviour
Nothing gets saved and many JavaScript functions on the page (e.g. block toolbar) don’t work anymore.

## WordPress information
- WordPress version: 5.7.0
- Gutenberg version: Not installed
- Are all plugins except Gutenberg deactivated? No
- Are you using a default theme (e.g. Twenty Twenty-One)? No

## Device information
- Device: Desktop
- Operating system: macOS 11.2.3 (20D91), Windows 10
- Browser: Google Chrome, Mozilla Firefox (both latest versions)

**Comment by posimyth** [comment]
@MatzeKitt We also having the same issue. It seems to because of 
`select("core/editor").isPreviewingPost()`
![image](https://user-images.githubusercontent.com/3997840/112295331-cfea2880-8cb9-11eb-81f5-f9e6ec2e73c8.png)

**Comment by jhume** [comment]
We are having the same issue.  Is there a workaround?  It's a major problem when pages can't be saved...

**Comment by MatzeKitt** [comment]
For me it has been fixed with the update to WordPress 5.7.

**Comment by Mamaduka** [comment]
Hello,

We tested this issue during the[ Editor Triage session today](https://wordpress.slack.com/archives/C02QB2JS7/p1627967697051500). While we couldn't reproduce the problem, it's clear that the `isPreviewingPost` selector could use some improvements.

Marking the issue as a bug, and I'll create PR shortly to handle cases when `options.isPreview` is undefined.
